internal/content_coding.go

```diff
@@ -408,7 +408,11 @@ func NewZstdDecoder() (*ZstdDecoder, error) {
 
 func (*ZstdDecoder) SetEncoding(string) {}
 
-func (d *ZstdDecoder) Decode(data []byte, _ int64) ([]byte, error) {
+func (d *ZstdDecoder) Decode(data []byte, maxDecompressionSize int64) ([]byte, error) {
+	size := int64(len(data))
+	if size > maxDecompressionSize {
```

Comment:
You need to check the size of the decoded data, not the original encoded data..

Reply:
gotcha.